Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h

```diff
@@ -308,6 +308,7 @@ for this property.
 - (void)_killWebContentProcess;
 - (void)_killWebContentProcessAndResetState;
 
+- (void)_saveResources:(NSURL *)directory suggestedFileName:(NSString *)name completionHandler:(void (^)(BOOL))completionHandler;
```

Comment:
Two comments.

1. We should avoid returning boolean for indication of success or failure. If it fails, that would be great if we could get the exact error. (ie. Permission error?, Disk full?, etc...)
2. About `suggestedFileName`. What if the suggested file name can't be used? Does it returns a failure, or it picks another filename. If the filename is changed, the caller should be notified.

About the sub resources, how do I know where are located? Is it a folder with with the very same name (minus the extension) ?

Reply:
> We should avoid returning boolean for indication of success or failure. If it fails, that would be great if we could get the exact error. (ie. Permission error?, Disk full?, etc...)

Sure, I could make the SPI return an NSError*.

> About suggestedFileName. What if the suggested file name can't be used? Does it returns a failure, or it picks another filename. If the filename is changed, the caller should be notified.

What do you mean the filename can't be used? Do you mean if it's malformed or file with same name already exists?

> About the sub resources, how do I know where are located? Is it a folder with the very same name (minus the extension) ?
By default, the subresources are stored in suggestedFileName_files, e.g. if you pass "apple", then in the directory, you should have file apple.html and directory apple_files that holds subresources.

Reply:
> What do you mean by "filename can't be used"? Do you mean the name is malformed or file with same name already exists?

If I pass "apple.html" for the suggestedFileName. What happens if the file already exists? Does it become "apple (1).html"? Does it fail? What if the "apple_files" exists?

Reply:
Hmm with current implementation we will fill the extension, so you should pass "apple", and we will save it as "apple.html". The operation can succeed and the existing files will be overwritten.